### PR TITLE
[WIP][Training - My] - Training research ajax form

### PR DESCRIPTION
**Pull request by truong-my-mulodo**
## Task list
- [x] Create post
- [x] Update post
- [x] Delete post
- [x] Show messages error validation

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -28,20 +29,23 @@ def update
     respond_to do |format|
       if @article.update(post_params)
         format.html { redirect_to articles_url , notice: 'Article has been updated ' }
```
Comment (nguyen-dat-mulodo) [inline app/controllers/articles_controller.rb]: Please remove this line if it is unnecessary

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -28,20 +29,23 @@ def update
...
     end
   end
 
   def destroy
-    @article.destroy
-    flash[:notice] = "Article has been deleted."
-    redirect_to articles_url
+    respond_to do |format|
+      if @article.destroy
+        format.html { redirect_to articles_url , notice: 'Article has been deleted.' }
```
Comment (nguyen-dat-mulodo) [inline app/controllers/articles_controller.rb]: Please remove this line if it is unnecessary

File: app/views/articles/_show.html.slim [hunk 3]
```diff
@@ -0,0 +1,15 @@
+p#notice
```
Comment (nguyen-dat-mulodo) [inline app/views/articles/_show.html.slim]: It is exist at application.html.slim page, please check it again.

File: app/views/articles/index.html.slim [hunk 4]
```diff
@@ -1,5 +1,7 @@
 h1 = 'Articles list'
-= link_to 'Create Article', new_article_path , class: "btn btn-primary", style: "margin-bottom:10px"
+= link_to 'Create Article', new_article_path, remote: true , class: "btn btn-primary", style: "margin-bottom:10px"
+div.row
+  div.col-md-6.col-md-offset-2.margin-article-bottom id="article-form" style="display:none;"
```
Comment (nguyen-dat-mulodo) [inline app/views/articles/index.html.slim]: Style of elements should be added in Css file.

File: app/assets/stylesheets/articles.scss [hunk 5]
```diff
@@ -5,3 +5,6 @@
 .margin-article{
   margin-left: 10px;
 }
+.margin-article-bottom{
```
Comment (nguyen-dat-mulodo) [inline app/assets/stylesheets/articles.scss]: Add blank line between div elements.
Add space same as ".margin-article-bottom {".

**Comment by truong-my-mulodo** [comment]
Hi @nguyen-dat-mulodo 
I fixed comment, please check them.
Thanks

**Comment by tran-hai-mulodo** [comment]
@truong-my-mulodo 

I reviewed. It's OK now. Thanks for your hard work.

**Pull request merged**